homeassistant/components/voicerss/tts.py

```diff
@@ -155,6 +260,7 @@
         vol.Optional(CONF_LANG, default=DEFAULT_LANG): vol.In(SUPPORT_LANGUAGES),
         vol.Optional(CONF_CODEC, default=DEFAULT_CODEC): vol.In(SUPPORT_CODECS),
         vol.Optional(CONF_FORMAT, default=DEFAULT_FORMAT): vol.In(SUPPORT_FORMATS),
+        vol.Optional(CONF_VOICE, default=DEFAULT_VOICE): vol.In(SUPPORT_VOICES),
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service, and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shifts in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
OK that's interesting to know (after two months of waiting and hoping).

So that means this great integration will stay in current shape until someone starts to migrate it to a config flow version?

Unfortunately, that's for sure way out of my skill level.

Too bad, I thought spending some hours solving a problem I have and making the improvement usable for everyone would be a rather quick and helpful thing. Unfortunately with all the needed rework I won't be able to work on this any further. Probably this integration will die (no maintainer, no huge user base, zero changes for years... one breaking change in core and goodbye VoiceRSS 😔).

Anyway, it was interesting to see and learn a bit how things work in the developers building. I'll move back to the users building, enough PR experience for a long time.